### PR TITLE
Don't show third-party JS errors in production builds

### PR DESCRIPTION
**Pull request by silverwind**
So we don't get issues like https://github.com/go-gitea/gitea/issues/29080 and https://github.com/go-gitea/gitea/issues/29273 any more. Only active in [production builds](https://webpack.js.org/guides/production/#specify-the-mode), in non-production the errors will still show.

### CODE REVIEW

**Comment by wxiaoguang** [comment]
I would suggest to check window.config.runModeIsProd and maybe also check whether current user is admin (in case the site admin tries to help users to figure out why "something doesn't work")

**Comment by silverwind** [comment]
`runModeIsProd` seems good, but the admin thing I wouldn't do, it's too strange behaviour do so something different per user role and admins can also have broken extensions installed.

**Comment by silverwind** [comment]
Now uses `runModeIsProd`.

**Review by wxiaoguang — APPROVED** [review]

**Review by 6543 — APPROVED** [review]

File: web_src/js/bootstrap.js [hunk 1]
```diff
@@ -29,17 +29,26 @@ export function showGlobalErrorMessage(msg) {
  * @param {ErrorEvent} e
  */
 function processWindowErrorEvent(e) {
+  const err = e.error ?? e.reason;
+  const assetBaseUrl = String(new URL(__webpack_public_path__, window.location.origin));
+
+  // error is likely from browser extension or inline script. Do not show these in production builds.
+  if (!err.stack?.includes(assetBaseUrl) && (window.config?.runModeIsProd ?? true)) return;
```
Comment (delvh) [inline web_src/js/bootstrap.js]: `?? true`?
Why?
If we are not sure, show the user errors

**Review by delvh — APPROVED** [review]

File: web_src/js/bootstrap.js (hunk 1 above)
Comment (silverwind) [reply web_src/js/bootstrap.js]: I defaulted it to hide errors when `window.config` is not there for whatever reason. It fits the style of this file where the other check for the variable is also with `?.` and there is a specific error message related to `window.config` being absent so we don't want to run into this error before that error message was emitted.

Comment (silverwind) [reply web_src/js/bootstrap.js]: Ah I see what you mean now. I guess we can default to showing it.

**Comment by silverwind** [comment]
Merging as I think @delv would approve latest change

**Comment by delvh** [comment]
I assume you mean me, and yes, I already approved

**Pull request merged**